api/tests/unit/segments/test_unit_segments_services.py

```diff
@@ -0,0 +1,265 @@
+from unittest.mock import MagicMock, patch
```

Comment:
FYI, I gave Agent another try following our meeting. 
This has been generated by Claude3.7.
It needed a couple of adjustments (errors in mocking, couple of wrong imports, missing some model hooks behavior..).
15-20minutes to clean-up so overall beneficial here